### PR TITLE
Adds Pyrsia and Rust blog

### PR DESCRIPTION
**Pull request by betarelease**
## Description
This PR does adds Pyrsia and Rust blog

## PR Checklist
- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### CODE REVIEW

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Why Pyrsia is being built in Rust?
+
+When we started working on Pyrsia we had the difficult and exciting task of choosing a language that would work for building secure supply chain software. We will share some details of how we decided to build Pyrsia in Rust and how it fits the problem we are set to solve.
+
+## Securing Open Source Software
+
+Open source software is built mostly by people passionate about solving a problem and sharing their solutions widely. One of the differences in how we build proprietary software vs open source is that, we find that many of those original developers do their best in keeping the software uptodate and try to patch vulnerabilities as soon as they can. Although their efforts are usually the best they can do. In some well publicized cases, developers have experienced burnout and lack of interest in supporting what they built, due to resource constraints.
+
+For proprietary software there are well published processes and patterns which are used to build software and record how it was done. Open Source software does not get this rigor and often is found vulnerable.
```
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:9]: s/For proprietary software /For proprietary software, `

Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:9]: `s/Open Source software does not/Open Source software usually does not`

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+On Pyrsia we focus on this missing piece of building open source software and are building a platform that offers build from source service while providing a record of how it was done. Pyrsia leverages a peer-to-peer distribution model for these binaries thus making the network resilient to failures.
+
+### Secure Software needs a secure platform
+
+While the aim of Pyrsia is to secure the software that it builds, a lot of trust/community involvement expects the network itself to be secure. Pyrsia as a platform has taken this expectation seriously and from the initial days invested a lot of energy into building it right.
+
+A few other considerations that were made during the initial discussions of Pyrsia include
```
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:17]: append `:`

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+* Decentralized network (think Web3) to leverage distribution of binaries across regions
+* Build from source using independent randomly chosen nodes to ensure security by reducing surface of attack
+* Consensus mechanism to ensure that multiple nodes participate in the build and verification process
+* Deploying Pyrsia node instances on wide variety of architectures, operating systems, as well as footprints(think Intel Xeon all the way to Raspberry * Pi and beyond)
```
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:22]: `s/footprints(think Intel Xeon all the way to Raspberry * Pi and beyond)/footprints (think Intel Xeon all the way to Raspberry Pi and beyond)`

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+> Rust began as a side project of Graydon Hoare, an employee at Mozilla. In short order, Mozilla saw the potential of the new language and began sponsoring it, before revealing it to the world in 2010.
+One possible source of the name, according to Hoare, is the rust fungus. This has caused Rust programmers to adopt “Rustaceans” as their moniker of choice.
+>
+> Despite its relative youth, Rust has steadily risen in the ranks of popular programming languages. In fact, while it ranked 33 in July 2019, by July 2020 it had risen to the 18th spot on the [TIOBE Programming Community Index](<https://www.tiobe.com/tiobe-index/>). Similarly, according to [Stack Overflow Developer Survey](https://insights.stackoverflow.com/survey/2020#technology-most-loved-dreaded-and-wanted-languages-loved), Rust has been the “most loved” language since 2016.
+
+### Rust language ecosystem
+
+The above philosophy made Rust a great candidate for use in security solutions like Pyrsia. Some other aspects that sealed the deal as a programming language for us were:
+
+* Performance close to quivalent C level programs [3]
```
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:61]: `s/quivalent/equivalent`

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+* Concurrent programming without the garbage collection [4]
+* Rust has a borrow checker which ensures references do not outlive the data
+* Rust can be compiled to reduced instruction set architectures
+
+Along with the above we also found that the initial set of libraries(libp2p, AlephBFT) we were looking to support had mature implementations in rust. Also we found that the communities that supported these libraries were welcoming all implementers and learners alike. This openness within the rust community in general made the choice easier for us.
```
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:66]: `s/Along with the above we also found that the initial set of libraries(libp2p, AlephBFT) we were looking to support had mature implementations in rust./Along with the above, we also found that the initial set of libraries (libp2p, AlephBFT) we were looking to support, had mature implementations in Rust.`

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+A lot of these appealing features of the Rust ecosystem are well summarized in [5].
+
+## Summary
+
+When we set out to change how open source software is secured we had a choice to make - the language to build the security solution with. When we surveyed what was available there were multiple options. C due to its performance, Golang because of its mature installations in the wild, Rust as an up and coming community with performance and modern language features.
+
+In the end it was clear to us that Rust was the right choice to make and we have started building Pyrsia in Rust. We realize that we have a steep learning curve and we are learning as a group. Come join us on our slack channel to discuss more.
```
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:73]: `s/slack channel/Slack channel` ? Also provide a link?

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md (hunk 5 above)
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:66]: `s/rust community/Rust community`

File: blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+### References
+
+* [Rust Philosophy](https://blog.rust-lang.org/2015/05/15/Rust-1.0.html)
+* [Rust History](https://www.talentopia.com/news/the-rust-programming-language-its-history-and-why/)
+* [Rust vs C](https://codilime.com/blog/rust-vs-c-safety-and-performance-in-low-level-network-programming/)
+* [Garbage collection issues](https://discord.com/blog/why-discord-is-switching-from-go-to-rust)
+* [Why projects use RUST?](https://codilime.com/blog/why-is-rust-programming-language-so-popular/#:~:text=High%20performance%20and%20safety%20are,amounts%20of%20data%20very%20quickly)
```
Comment (jperedadnr) [inline blog/2022-09-24-why-pyrsia-is-being-built-in-rust.md:81]: `s/RUST/Rust`

**Review by jperedadnr — COMMENTED** [review]
Looks good, some minor formatting comments

**Comment by betarelease** [comment]
Thanks for the review comments. Although I did it here I think it is better in the website repo - because the next blog has images and that will now start polluting the core codebase.

**Comment by betarelease** [comment]
Addressed all your comments and updated blog on https://github.com/pyrsia/pyrsia.github.io/pull/125

**Pull request closed without merging**